exam/models.py

```diff
@@ -88,41 +90,37 @@ def assign(self):
         """
 
         exam_rooms = ExamRoom.objects.filter(available=True)
-        exam_students = Student.get_current(tutor_group__group_category=self.group_category, want_exam=True).order_by('name', 'prename')
-        student_count = len(exam_students)
+        exam_students = Student.get_current(tutor_group__group_category=self.group_category, want_exam=True)
+        exam_students = exam_students.order_by('name', 'prename')
+        student_count = exam_students.count()
 
-        if len(exam_rooms) == 0 or student_count == 0:
+        if exam_rooms.count() == 0 or student_count == 0:
             return 0
 
-        free_places = sum([exam_room.capacity(self.spacing) for exam_room in exam_rooms])
-        places_needed = len(exam_students)
+        aggregation_string = 'capacity_{}_free'.format(self.spacing)
 
-        # Set ratio either to a value where all rooms are used or
-        # fixed 0.9 to fill up the first rooms up to 90 percent each
-        ratio = places_needed / free_places if self.mode == 0 else 0.9
+        total_places = exam_rooms.aggregate(total_places=Sum(aggregation_string)).get('total_places')
 
-        total_used_places = 0
-        split_points = []
+        # Set ratio so that all room gets equals percentage
+        ratio = student_count / total_places
+
+        # on minimal room mode and if the ratio is less then 90% the ratio is set to 90%
+        # e.g. the first rooms get filled by 90%. The other rooms are not used
+        if self.mode == 1 and ratio < 0.9:
```

Comment:
It would be nice if the `assign` method would use enums for its modes, but this is out of scope of this pull request.